README.md

```diff
@@ -41,96 +73,185 @@ pod "KlaviyoSwift"
-
-After the SDK is installed you can begin tracking events in your app.
+</details>
 
-1. Make sure any .swift files using the Klaviyo SDK contain the following import call:
+## Initialization
+The SDK must be initialized with the short alphanumeric [public API key](https://help.klaviyo.com/hc/en-us/articles/115005062267#difference-between-public-and-private-api-keys1)
+for your Klaviyo account, also known as your Site ID.
 
 ```swift
+// AppDelegate
+
 import KlaviyoSwift
+
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        KlaviyoSDK().initialize(with: "YOUR_KLAVIYO_PUBLIC_API_KEY")
+        return true
+    }
+}
 ```
 
-2. To add Klaviyo's tracking functionality, include the following line in AppDelegate.swift, within `application:didFinishLaunchingWithOptions`:
+The SDK **should** be initialized before any other Klaviyo SDK methods are called.
+
+## Profile Identification
+The SDK provides methods to identify your users as Klaviyo profiles via the [Create Client Profile API](https://developers.klaviyo.com/en/reference/create_client_profile).
+A profile can be identified by any combination of the following:
+
+* External ID: A unique identifier used by customers to associate Klaviyo profiles with profiles in an external system, such as a point-of-sale system. Format varies based on the external system.
+* Individual's email address
+* Individual's phone number in [E.164 format](https://help.klaviyo.com/hc/en-us/articles/360046055671#h_01HE5ZYJEAHZKY6WZW7BAD36BG)
+
+These above Identifiers are persisted to local storage so that the SDK can keep track of the current user/profile for you when you make event requests or want to set a push token etc.
+
+Profile identifiers can be set all at once or individually. Either way, the SDK will group and batch API calls to improve performance.
+
+The following code demonstrates how to set profile identifiers:
 
 ```swift
-KlaviyoSDK().initialize(with: "YOUR_KLAVIYO_PUBLIC_API_KEY")
+// organization, title, image, location and additional properties (dictionary) can also be set using the below constructor
+let profile = Profile(email: "junior@blob.com",  firstName: "Blob",  lastName: "Jr.")
+KlaviyoSDK().set(profile: profile)
+
+// or setting individual properties
+KlaviyoSDK().set(profileAttribute: .firstName, value: "Blob")
+KlaviyoSDK().set(profileAttribute: .lastName, value: "Jr.")
 ```
 
-3. Begin tracking events anywhere within your application by calling the `create(event:)` method in the relevant location.
+### Reset Profile
+To start a new profile altogether (e.g. if a user logs out) either call `KlaviyoSDK().resetProfile()` to clear the currently tracked profile identifiers,
+or use `KlaviyoSDK().set(profile: profile)` to overwrite it with a new profile object.
+<TODO: Should we talk about what happens to a push token here?>
```

Comment:
Probalbly either here or in the section about push tokens.

Reply:
Ya looks like this is actually already covered by the `Push Tokens And Multiple Profiles` section? Maybe can just remove this TODO?

Reply:
Yeah I added this TODO before adding that section. I'll remove the todo.